[WIP]feat: write and delete tuples from json file

gabrielbussolo opened the pull request:
<!-- Thanks for opening a PR! Here are some quick tips:
If this is your first time contributing, [read our Contributing Guidelines](https://github.com/openfga/.github/blob/main/CONTRIBUTING.md) to learn how to create an acceptable PR for this repo.
By submitting a PR to this repository, you agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md)

If your PR is under active development, please submit it as a "draft". Once it's ready, open it up for review.
-->

aims to fix #161 

## Description
<!-- Provide a detailed description of the changes -->

## References
<!-- Provide a list of any applicable references here (Github Issue, [OpenFGA RFC](https://github.com/openfga/rfcs), other PRs, etc..) -->

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

gabrielbussolo commented:
if you can take a look @rhamzeh ;D

rhamzeh reviewed (commented):
Thanks! Can I also ask you to run `make format` and `make lint` to make sure we got rid of any lint errors?

rhamzeh commented:
@gabrielbussolo may I ask you to also update the README in this PR to document the new options for write and delete?

gabrielbussolo commented:
> @gabrielbussolo may I ask you to also update the README in this PR to document the new options for write and delete?

sure thing @rhamzeh, ill do it tomorrow ;D

gabrielbussolo commented on cmd/tuple/write.go:
```diff
@@ -32,13 +34,53 @@ var writeCmd = &cobra.Command{
 	Short:   "Create Relationship Tuples",
 	Long:    "Add relationship tuples to the store.",
 	Example: "fga tuple write --store-id=01H0H015178Y2V4CX10C2KGHF4 user:anne can_view document:roadmap",
-	Args:    cobra.ExactArgs(3), //nolint:gomnd
```
do you mean for the whole write/delete command? or just the args validation?

gabrielbussolo replied:
about the args validation: https://github.com/openfga/cli/pull/165#discussion_r1338353921

gabrielbussolo commented on cmd/tuple/delete.go:
```diff
@@ -32,13 +33,53 @@ var deleteCmd = &cobra.Command{
 	Short:   "Delete Relationship Tuples",
 	Long:    "Delete relationship tuples from the store.",
 	Example: "fga tuple delete --store-id=01H0H015178Y2V4CX10C2KGHF4 user:anne can_view document:roadmap",
-	Args:    cobra.ExactArgs(3), //nolint:gomnd
```
because we need 3 args OR the --file i created a custom function for it to be used on the `Args` field: https://github.com/openfga/cli/blob/0b0ec8d217c03edfe0863e50a1ee7e7727826642/cmd/tuple/delete.go#L108-L116

rhamzeh replied:
Thank you!

gabrielbussolo commented:
> @gabrielbussolo may I ask you to also update the README in this PR to document the new options for write and delete?

Updated the README, also fixed the comments. ;D

There is a specific lint _screaming_ at me:
```
➜  openfga-cli git:(161-add-support-for-specifying-a-list-of-tuples-to-delete) ✗ make lint
go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest && golangci-lint run
# github.com/golangci/golangci-lint/cmd/golangci-lint
ld: warning: -bind_at_load is deprecated on macOS
cmd/tuple/delete.go:111:11: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"at least %d arg(s) are required OR the flag --%s\", n, flag)" (goerr113)
                        return fmt.Errorf("at least %d arg(s) are required OR the flag --%s", n, flag)
                               ^
make: *** [lint] Error 1
```

is asking to create a specific error for it, but on this case I dont think it makes sense once i need to format the error before I send it (same approach as the `cobra.ExactArgs`). 

also, about the lint, it forces to create a const for a number, and its ok on some contexts, but when we have a function such as `cobra.ExactArgs(3)` its pretty explicit what it does, do imho we could ignore it.

the first one i didnt solved, the second one i created a const. What is the team approach on it?

rhamzeh commented:
> the first one i didnt solved, the second one i created a const. What is the team approach on it?

For both of these cases, it's safe to add a lint ignore.

- `//nolint:goerr113` for the error (See here https://github.com/openfga/cli/blob/cb4adfe7470e793748d4278075ce96ba755be404/cmd/store/create.go#L62-L61)
- `//nolint:gomnd` for the const (See here https://github.com/openfga/cli/blob/48eb34cf8ece4e2e5d518748c0fa9a1cca7e8815/cmd/query/check.go#L58)

gabrielbussolo commented:
> Other than this lint issue, good to merge!

perfect, so its done @rhamzeh

rhamzeh commented:
Thanks @gabrielbussolo!
